rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll

```diff
@@ -245,6 +312,20 @@ private module Cached {
     i = -1
   }
 
+  /**
+   * Holds if the call `call` at index `i` in basic block `bb` may reach a callable
+   * that writes captured variable `v`.
+   */
+  cached
+  predicate capturedCallWrite(CallExprBase call, BasicBlock bb, int i, Variable v) {
```

Comment:
There should be no need to cache this.

Reply:
Should I just remove the `cached` annotation or should I also move it out of the `Cached` module?

Reply:
The [Ruby equivalent](https://github.com/paldepind/codeql/blob/cab916453d1cd6ce6dd9c41c716474c12c3f770d/ruby/ql/lib/codeql/ruby/dataflow/internal/SsaImpl.qll?plain=1#L322) is also cached. Is there something different about that one that causes it to need caching?

Reply:
It needs to be moved out, otherwise the compiler will complain (all public declarations in a `cached` module must be marked as `cached`).

The reason why it is cached in Ruby is that it is referenced in a user-visible class.

Reply:
I see, thanks. I've addressed the comments :)